### DISCUSSION TITLE
Welcome to ModbusSniffer Discussions!

### DISCUSSION THREAD

**Discussion by snhobbs** [snhobbs/ModbusSniffer#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by PerfoPerfo** [comment]
Hello, This is perfect for what I've been looking for thanks.  I have no documentation on a Modbus device I've got though it is communicating fine between it and the master.  Am I right in thinking as my Modbus is rs485 I need an rs485 to usb converter then wire the rs485 end of the converter to the slave A and B terminals (i.e. in parallel with the existing link).  I can then run this on the windows machine and see the traffic being sent back and forth between the master and slave ?

**Reply by snhobbs** [reply]
If your device has an A/B as the serial names then it's almost guaranteed to be half duplex RS-485 like you say. This lib will work but I would recommend getting a cheap logic analyzer and using the modbus decoder stacks with pulseview/sigrok. Here's a blog post on doing that: https://www.maskset.net/blog/2022/10/13/sigrok--10-logic-analyzer/

You can also use an ftdi usb-rs485 based dongle with pulseview/sigrok.

**Reply by PerfoPerfo** [reply]
Thanks for the reply. Using the logic analyser can I see the commands in print ? Ie not just the binary train ?   Why do you think this will be better than your lib ?  I haven’t got the hardware yet so could order either or both…. Thanks again
